osi_featuredata.proto

```diff
@@ -316,6 +400,154 @@ message LidarDetection
+// signals from other ultrasonic sensors are included in this message.
+//
+// Direct detections:
+//
+// Sending: Ultrasonic Sensor ID 1
+// Receiving: Ultrasonic Sensor ID 1
+// \image html OSI_USSensor_direct.png
+//
+// Indirect detections:
+//
+// Sending: Ultrasonic Sensor ID 1
+// Receiving: Ultrasonic Sensor ID 2 or/and 3
+// \image html OSI_USSensor_indirect.png
+//
+// \note required for USS
+//
+message UltrasonicDetectionData
+{
+    // Header attributes of ultrasonic detection from one ultrasonic sensor.
+    //
+    optional SensorDetectionHeader header = 1;
+
+    // Additional header attributes of ultrasonic detection from one ultrasonic
+    // sensor.
+    //
+    optional UltrasonicDetectionSpecificHeader specific_header = 3;
+
+    // List of ultrasonic detections.
+    //
+    // \note OSI uses singular instead of plural for repeated field names.
+    //
+    repeated UltrasonicDetection detection = 2;
+
+    // List of ultrasonic indirect detections (sender and receiver sensors are
+    // not the same).
+    //
+    // \note OSI uses singular instead of plural for repeated field names.
+    //
+    repeated UltrasonicIndirectDetection indirect_detection = 4;
+}
+
+//
+// \brief Ultrasonic detection from the sensor (same sensor as sender and
+// receiver).
+//
+// Direct detections:
+//
+// Sending: Ultrasonic Sensor ID 1
+// Receiving: Ultrasonic Sensor ID 1
+// \image html OSI_USSensor_direct.png
+//
+message UltrasonicDetection
+{
+    // Existence probability of the detection not based on history. Value does
+    // not depend on any past experience with similar detections.
+    //
+    // \note Used as confidence measure where a low value means less confidence
+    // and a high value indicates strong confidence.
+    //
+    optional double existence_probability = 1;
+
+    // ID of the detected object this detection is associated to.
+    //
+    // \note ID = MAX(uint64) indicates no reference to an object.
+    //
+    optional Identifier object_id = 2;
+
+    // Measured distance (radius) of the detection.
+    //
+    // Unit: [m]
+    //
+    optional double distance = 3;
+}
+
+//
+// \brief Ultrasonic detection received by an other ultrasonic sensor (different
+// sensors as sender and receiver).
+//
+// Indirect detections:
+//
+// Sending: Ultrasonic Sensor ID 1
+// Receiving: Ultrasonic Sensor ID 2 or/and 3
+// \image html OSI_USSensor_indirect.png
+//
+message UltrasonicIndirectDetection
+{
+    // Existence probability of the detection not based on history. Value does
+    // not depend on any past experience with similar detections.
+    //
+    // \note Used as confidence measure where a low value means less confidence
+    // and a high value indicates strong confidence.
+    //
+    optional double existence_probability = 1;
+
+    // ID of the detected object this detection is associated to.
+    //
+    // \note ID = MAX(uint64) indicates no reference to an object.
```

Comment:
invalid IDs should be treated the same way in all OSI ID messages? Please add issue to check all other IDs